### PR TITLE
Initial Release

### PR DESCRIPTION
**Pull request by elian-kurtenkov**

### CODE REVIEW

File: src/Lykke.blue.Api/Controllers/PledgesController.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        {
+            if (request == null)
+                return BadRequest();
+
+            var clientRequest = Mapper.Map<ClientModel.CreatePledgeRequest>(request);
+            clientRequest.ClientId = _requestContext.ClientId;
+
+            var pledgeApiResponse = await _pledgesApi.CreatePledgeWithHttpMessagesAsync(clientRequest);
+
+            var badResponse = await pledgeApiResponse.CheckClientResponseForErrors();
```
Comment (samodovdi) [inline src/Lykke.blue.Api/Controllers/PledgesController.cs:60]: check pledgeApiResponse for null

File: src/Lykke.blue.Api/Controllers/PledgesController.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        [HttpGet]
+        [SwaggerOperation("GetPledge")]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [ProducesResponseType(typeof(ApiResponses.GetPledgeResponse), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> Get()
+        {
+            var pledgeApiResponse = await _pledgesApi.GetPledgeWithHttpMessagesAsync(_requestContext.ClientId);
+
+            var badResponse = await pledgeApiResponse.CheckClientResponseForErrors();
```
Comment (samodovdi) [inline src/Lykke.blue.Api/Controllers/PledgesController.cs:82]: check pledgeApiResponse for null

File: src/Lykke.blue.Api/Controllers/PledgesController.cs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        {
+            if (request == null)
+                return BadRequest();
+
+            var clientRequest = Mapper.Map<ClientModel.UpdatePledgeRequest>(request);
+            clientRequest.ClientId = _requestContext.ClientId;
+
+            var pledgeApiResponse = await _pledgesApi.UpdatePledgeWithHttpMessagesAsync(clientRequest);
+
+            var badResponse = await pledgeApiResponse.CheckClientResponseForErrors();
```
Comment (samodovdi) [inline src/Lykke.blue.Api/Controllers/PledgesController.cs:115]: the same here

File: src/Lykke.blue.Api/Controllers/PledgesController.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        [HttpDelete]
+        [SwaggerOperation("DeletePledge")]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [ProducesResponseType(typeof(void), (int)HttpStatusCode.NoContent)]
+        public async Task<IActionResult> Delete()
+        {
+            var pledgeApiResponse = await _pledgesApi.DeletePledgeWithHttpMessagesAsync(_requestContext.ClientId);
+
+            var badResponse = await pledgeApiResponse.CheckClientResponseForErrors();
```
Comment (samodovdi) [inline src/Lykke.blue.Api/Controllers/PledgesController.cs:136]: and here

File: src/Lykke.blue.Api/Controllers/RefLinksController.cs [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+
+            var result = await ExecuteRefLinksMethod((p) => _referralLinksService.FinalizeRefLinkTransferWithHttpMessagesAsync(p), serviceRequest, request.LogMessage);
+            return result;         
+        }
+
+        private async Task<IActionResult> ExecuteRefLinksMethod<U, T>(Func<U, Task<HttpOperationResponse<T>>> method, U request, string logMessage)
+        {
+            var response = await method(request);
+
+            var error = await response.CheckClientResponseForErrors();
```
Comment (samodovdi) [inline src/Lykke.blue.Api/Controllers/RefLinksController.cs:204]: check response for null here aswell

File: src/Lykke.blue.Api/Controllers/RefLinksController.cs [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+//    {
+//        return error;
+//    }
+
+//    var result = response.Body;
+
+//    await LogInfo(request, ControllerContext, $"Finalize offchain transfer: {new { response.Response.StatusCode, result }.ToJson()}");
+
+//    return StatusCode((int)response.Response.StatusCode, result);
+//}
```
Comment (samodovdi) [inline src/Lykke.blue.Api/Controllers/RefLinksController.cs]: unused code can be removed instead of commented out

**Review by samodovdi — COMMENTED** [review]
all unused commented out code can be removed

File: src/Lykke.blue.Api/Controllers/PledgesController.cs (hunk 1 above)
Comment (filip-prime) [reply src/Lykke.blue.Api/Controllers/PledgesController.cs:60]: null check is now handled in CheckClientResponseForErrors

File: src/Lykke.blue.Api/Controllers/PledgesController.cs (hunk 2 above)
Comment (filip-prime) [reply src/Lykke.blue.Api/Controllers/PledgesController.cs:82]: null check is now handled in CheckClientResponseForErrors

File: src/Lykke.blue.Api/Controllers/PledgesController.cs (hunk 3 above)
Comment (filip-prime) [reply src/Lykke.blue.Api/Controllers/PledgesController.cs:115]: null check is now handled in CheckClientResponseForErrors

File: src/Lykke.blue.Api/Controllers/PledgesController.cs (hunk 4 above)
Comment (filip-prime) [reply src/Lykke.blue.Api/Controllers/PledgesController.cs:136]: null check is now handled in CheckClientResponseForErrors

File: src/Lykke.blue.Api/Controllers/RefLinksController.cs (hunk 5 above)
Comment (filip-prime) [reply src/Lykke.blue.Api/Controllers/RefLinksController.cs:204]: null check is now handled in CheckClientResponseForErrors

**Comment by filip-prime** [comment]
null check is now handled in CheckClientResponseForErrors

**Comment by filip-prime** [comment]
CheckClientResponseForErrors is an extension method  of HttpOperationResponse and allows to be called if pledgeApiResponse is null. However, null check must be performed inside CheckClientResponseForErrors  before using HttpOperationResponse param. So this is where I put it.

**Review by samodovdi — APPROVED** [review]

**Pull request merged**